test/InterversionTests/RavenDB-20628-Interversion.cs

```diff
@@ -0,0 +1,142 @@
+namespace SlowTests.Issues
+{
+    public class RavenDB_20628_Interversion : MixedClusterTestBase
+    {
+        public RavenDB_20628_Interversion(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task ClusterTransaction_Failover_Shouldnt_Throw_ConcurrencyException()
+        {
+            var (leader, peers, local) = await CreateMixedCluster(new[]
+            {
+                "5.4.105",
+                "5.4.105"
+            });
+
+            string databaseName = GetDatabaseName();
+            var urls = new List<string>(){ leader.WebUrl }.Concat(peers.Select(l => l.Url)).ToArray();
+
+            var tags = leader.ServerStore.GetClusterTopology().AllNodes.Keys.Where(x => x != leader.ServerStore.NodeTag).ToList();
+
+
+            using (var store1 = new DocumentStore { Urls = new[]
+                   {
+                       leader.WebUrl
+                   } }.Initialize())
+            {
+                var record = new DatabaseRecord(databaseName)
+                {
+                    Topology = new DatabaseTopology
+                    {
+                        Members = new List<string>
+                        {
+                            leader.ServerStore.NodeTag,
+                        }.Concat(tags).ToList()
+                    },
+                    Settings =
+                    {
+                        [RavenConfiguration.GetKey(x => x.Replication.ReplicationMinimalHeartbeat)] = "1",
+                        [RavenConfiguration.GetKey(x => x.Replication.RetryReplicateAfter)] = "1",
+                        [RavenConfiguration.GetKey(x => x.Core.RunInMemory)] = true.ToString(),
+                        [RavenConfiguration.GetKey(x => x.Core.ThrowIfAnyIndexCannotBeOpened)] = true.ToString(),
+                        [RavenConfiguration.GetKey(x => x.Indexing.MinNumberOfMapAttemptsAfterWhichBatchWillBeCanceledIfRunningLowOnMemory)] = int.MaxValue.ToString()
+                    }
+                };
+                await CreateDatabase(store1, 3, dbName: databaseName, record);
+            }
+
+
+            using var store = new DocumentStore
+            {
+                Urls = new [] { leader.WebUrl },
+                Database = databaseName
+            }.Initialize();
+
+            var killed = false;
+            
+            var disposeNodeTask = Task.Run(async () =>
+            {
+                await Task.Delay(700);
+                var url = store.GetRequestExecutor(databaseName).TopologyNodes.First().Url;
+
+                Assert.Equal(leader.WebUrl, url);
+                await DisposeServerAndWaitForFinishOfDisposalAsync(leader);
+                killed = true;
+            });
+            await ProcessDocument(store, "Docs/1-A");
+
+            await disposeNodeTask;
+            Assert.True(killed);
+
+            foreach (var p in peers)
+            {
+                await KillSlavedServerProcessAsync(p.Process);
+            }
```

Comment:
shouldn't this be part of the test dispose already?

Reply:
No

Reply:
why not?